provider/provider.go

```diff
@@ -98,3 +102,16 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 
 	return l, diags
 }
+
+func leveledLog(level string) func(format string, v ...interface{}) {
```

Comment:
Thanks for the proper log implementation, which follows the terraform best-practices 👍